calendar/src/main/java/com/minuk/calendar/PetCalendarView.kt

```diff
@@ -0,0 +1,304 @@
+        }
+    }
+
+    /**
+     * 측정된 가로와 세로 길이로 ratio를 계산하여
+     * 이전 ratio와 다르면 item view 재생성
+     */
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+
+        when (MeasureSpec.getMode(widthMeasureSpec)) {
+            MeasureSpec.UNSPECIFIED -> viewWidth = widthMeasureSpec
+            MeasureSpec.EXACTLY -> viewWidth = MeasureSpec.getSize(widthMeasureSpec)
+        }
+
+        when (MeasureSpec.getMode(heightMeasureSpec)) {
+            MeasureSpec.UNSPECIFIED -> viewHeight = heightMeasureSpec
+            MeasureSpec.EXACTLY -> viewHeight = MeasureSpec.getSize(heightMeasureSpec)
+        }
+
+        setItemViewDimensRadio(true)
+    }
+
+    private fun setItemViewDimensRadio(update: Boolean = false) {
+        val ratio = getItemViewDimensRatio()
+
+        if (itemViewRatio != ratio) {
+            itemViewRatio = ratio
+
+            if (update) {
+                calendarAdapter?.notifyDataSetChanged()
+            }
+        }
+    }
+
+    /**
+     * ConstraintLayout DimensRatio 구하는 함수
+     *
+     * @return width:height  ex) 2:3
+     */
+    private fun getItemViewDimensRatio(): String {
+        var verticalAxisSize = daysCount / 7
+
+        val itemViewWidth = viewWidth / 7
+        val itemViewHeight = (viewHeight - headerViewHeight) / verticalAxisSize
+
+        val gcd = getGCD(itemViewWidth, itemViewHeight.toInt())
+        //Log.d(TAG, "width: $itemViewWidth, height: $itemViewHeight, gcd: $gcd")
+
+        return "${itemViewWidth / gcd}:${(itemViewHeight / gcd).toInt()}"
+    }
+
+    /**
+     * 유클리드 호제법으로 구하는 a와 b의 최대공약
+     */
+    private fun getGCD(a: Int, b: Int): Int {
+        var maximum = max(a, b)
+        var minimum = min(a, b)
+
+        while (minimum != 0) {
+            val r = maximum % minimum
+
+            if (r == 0) return minimum
+
+            maximum = b
+            minimum = r
+        }
+
+        return minimum
+    }
+
+    fun setCalendarDate(
+        calendar: Calendar
+    ) {
+        setCalendarDate(
+            calendar[Calendar.YEAR],
+            calendar[Calendar.MONTH],
+            calendar[Calendar.DAY_OF_MONTH]
+        )
+    }
+
+    fun setCalendarDate(
+        year: Int = calendar[Calendar.YEAR],
+        month: Int = calendar[Calendar.MONTH],
+        day: Int = calendar[Calendar.DAY_OF_MONTH]
+    ) {
+        calendar.set(year, month, day)
+
+        initCalendarData()
+    }
+
+    fun setEventHandler(eventHandler: PetCalendarEventHandler) {
+        this.eventHandler = eventHandler
+    }
+
+    private inner class PetCalendarAdapter :
+        RecyclerView.Adapter<PetCalendarAdapter.PetCalenderViewHolder>() {
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PetCalenderViewHolder {
+            val inflater = LayoutInflater.from(parent.context)
+            return PetCalenderViewHolder(ItemCalendarBinding.inflate(inflater))
```

Comment:
```suggestion
            val binding = ItemCalendarBinding.inflate(LayoutInflater.from(parent.context), parent,false)
        return PetCalenderViewHolder(binding)
        // or
        return PetCalenderViewHolder(
            ItemCalendarBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
```
안녕하세요. PR 잘 읽었습니다. 여기 inflate에서 ViewGroup과 attachToRoot를 생략하는  이유가 있을까요? 개인적으로 저는 위와같이 사용하는데 궁금해서 리뷰남깁니다.

Reply:
안녕하세요 :)

`ItemCalendarBinding.inflate(inflater)` 코드로도 작성했을 때 내부적으로 자식 ConstraintLayout의 `DimensRatio`를 사용해 뷰 크기 제약을 적용해서 결과물을 보았을 때 큰 문제점을 못 느낀 것 같아서 해당 코드로 작성한 듯 하네요..

하지만 피드백을 받아서 `inflate()`와 `inflate(inflater, viewGroup, attachToParent)의 차이점에 대해 공부해 보았습니다.

- **viewGroup** : 생성될 View의 Root를 명시해준다.  null일 경우에는 LayoutParams값을 설정 할 수 없기 때문에 XML내의 최상위 android:layout_xxxxx값들이 무시된다.

- **attachToRoot**: `true`로 설정할 경우 생성된 View가 Root 자식으로 자동으로 추가 된다.

현재 `item_calendar.xml` Root layout에 `android:layout_xxxx` 속성들이 많은 것은 아니지만, `infalte()`를 사용하면 뷰가 생성되기 직전에 `LayoutParam` 속성에 접근하면 `NPE`를 발생하는 듯하네요. 

말씀해주신`ItemCalendarBinding.inflate(LayoutInflater.from(parent.context), parent,false)` 방법을 사용하는 것이 유지보수 측면에서 상당히 좋을 듯 합니다. 감사합니다.